maps.go

```diff
@@ -0,0 +1,67 @@
+
+// Package maps defines various functions useful with maps of any type.
+// Note: Will be replaces by stdlib "package maps" in the future
+package golds
+
+// Keys returns the keys of the map m.
+// The keys will be an indeterminate order.
+func keys[K comparable, V any](m map[K]V) []K{
```

Comment:
Why not make the map a type itself? 🤔 
```
type Map[K comparable, V any] map[K]V

func (m Map[K, V]) Keys() []K {
	keys := make([]K, 0, len(m))
	for k := range m {
		keys = append(keys, k)
	}
	return keys
}
```

Reply:
Because it's a draft PR 😉 (also see link in the PR description)

Reply:
I see! Well, maybe functions are better after all, since some cannot be writtem as methods anyway (ones that do a type conversion of some sorts, like `Apply[K comparable, V1, V2 any](m1 Map[K, V1], fn func(V1) V2) Map[K, V2]`)